core/bin/external_node/src/config/mod.rs

```diff
@@ -242,6 +242,15 @@ pub struct OptionalENConfig {
     // This is intentionally not a part of `RemoteENConfig` because fetching this info from the main node would defeat
     // its purpose; the consistency checker assumes that the main node may provide false information.
     pub contracts_diamond_proxy_addr: Option<Address>,
+
+    #[serde(default = "OptionalENConfig::default_snapshots_recovery_enabled")]
+    pub snapshots_recovery_enabled: bool,
+
+    #[serde(default = "OptionalENConfig::default_pruning_chunk_size")]
+    pub pruning_chunk_size: u32,
+
+    /// If set, l1 batches will be pruned after they are that long
```

Comment:
Returning back to the command-line flag vs config argument: IMO, it's not very obvious that not setting this field should disable pruning (rather than, say setting this value to 0 or to some reasonable default). If we go with the config approach, it could make sense to have a boolean flag just to enable / disable pruning explicitly.

Reply:
I feel like we're spending relatively large amounts of time discussing such details and I think crucial is not how this is enabled, but how well we'll announce and document it in ansible repo and partners channel